src/helix/core.cljs

```diff
@@ -217,6 +217,39 @@
 
   (rds/renderToString ($$ MyComponent {:foo "baz"})))
 
+(defn ref
+  "Like react/createRef, but the ref can be swapped, reset, and dereferenced
+  like an atom.
+
+  Note: `helix.core/ref` is mostly used for class components. Function
+  components typically rely on `helix.hooks/use-ref` instead."
+
+  ([]
+   (ref nil))
+
+  ([initial-value]
+   (let [^js ref (react/createRef)]
+     (set! (.-current ref)
+           (specify! #js {:current initial-value}
```

Comment:
I can't remember why I don't call `specify!` on the return value of `use-ref` directly, but we could try that here and remove one layer of object nesting

Reply:
Aha that makes sense!